packages/salesforcedx-vscode-soql/src/client/client.ts

```diff
@@ -55,9 +70,29 @@ export function startLanguageClient(context: ExtensionContext): void {
     serverOptions,
     clientOptions
   );
+  client.registerFeature(new QueryValidationFeature());
```

Comment:
This feature sets `capabilities.soql.runQuery` so the server knows it's ok to send the `RunQuery` request for validation